clang/lib/CodeGen/CodeGenPGO.cpp

```diff
@@ -983,7 +979,7 @@ void CodeGenPGO::mapRegionCounters(const Decl *D) {
   // for most embedded applications. Setting a maximum value prevents the
   // bitmap footprint from growing too large without the user's knowledge. In
   // the future, this value could be adjusted with a command-line option.
-  unsigned MCDCMaxConditions = (CGM.getCodeGenOpts().MCDCCoverage) ? 6 : 0;
+  unsigned MCDCMaxConditions = (CGM.getCodeGenOpts().MCDCCoverage) ? 32767 : 0;
```

Comment:
I think the limit should be controlled by a flag, with a reasonable default value.

E.g.

`-mcdc-max-conditions=<value>`

Reply:
`MaxCond` is one of options. I guess it could be configurable just for theirs coding rules, since I think controlling `MaxCond` would not make sense.
I know we have to introduce other options as well. I will do later.
* `MaxCond`
* `MaxTVs` to restrict per-expression number of TVs
* "Max bytes of bitmap in the CU" to restict inter-function size of the bitmap.

Reply:
I know you will do this later, but I also wanted to echo the desire for a means to control the condition limit to warn the user against unintended memory footprint expansion, most useful in embedded development contexts.  It may not otherwise be obvious to most users what is contributing to larger memory usage. We could also override the default in downstream toolchains.

Reply:
Also why 32767? Max signed 16-bit value? Is there a reason?

Reply:
@evodius96 I've introduced `-fmcdc-max-conditions=32767`.
> against unintended memory footprint expansion
I think `-fmcdc-max-test-vectors=64` can restrict max bitmap size as the current implementation.

@ornata I have no idea how to determine practical limit for that. So I use `SHRT_MAX`.